File: Projekt/Frontend/system-kliniki-medycznej/src/app/features/prescriptions/model/prescription-details.ts
import {DoctorInfo} from '../../doctor/domain/doctor-info';
import {PrescriptionMedicineInfo} from './prescription-medicine-info';
import {PatientData} from '../../patient/model/patient-data';

export interface PrescriptionDetails {
  id: string;
  doctorInfo: DoctorInfo;
  description: string;
  createdAt: string;
  expirationDate: string;
  patient: PatientData;
  prescriptionMedicineInfoList: PrescriptionMedicineInfo[];
}

File: Projekt/Backend/system-kliniki-medycznej/src/main/java/org/back/systemklinikimedycznej/prescription/controller/dto/PrescriptionDetails.java
package org.back.systemklinikimedycznej.prescription.controller.dto;

import lombok.Builder;
import org.back.systemklinikimedycznej.cure.dto.PrescriptionMedicineInfo;
import org.back.systemklinikimedycznej.doctor.controller.dto.DoctorInfo;
import org.back.systemklinikimedycznej.patient.controllers.dto.PatientData;

import java.time.LocalDate;
import java.util.List;
import java.util.UUID;

@Builder
public record PrescriptionDetails(
        UUID id,
        DoctorInfo doctorInfo,
        String description,
        LocalDate createdAt,
        LocalDate expirationDate,
        PatientData patient,
        List<PrescriptionMedicineInfo> prescriptionMedicineInfoList

) {
}
